### PR TITLE
Make xrCompatible renderer option

### PR DESCRIPTION
**Pull request by mkeblx**
Addressing #19275 by making `xrCompatible` a parameter, and enable xr based off that.

(Would need to update examples.)

### CODE REVIEW

**Comment by mrdoob** [comment]
I don't think this is what @toji suggested... 🤔

**Comment by mrdoob** [comment]
https://github.com/mrdoob/three.js/issues/19275#issuecomment-626978061

**Comment by toji** [comment]
This is basically what I suggested, at least in part. I think it would be pretty sensible to also add a call to  `gl.makeXRCompatible()` as @mrdoob suggested in #19275 if this property isn't set up front. You should be able to tell if the context was made XR Compatible previously by checking the `xrCompatible` attribute of the dictionary returned by `gl.getContextCreationAttributes()`

File: src/renderers/webxr/WebXRManager.js [hunk 1]
```diff
@@ -181,6 +181,9 @@ function WebXRManager( renderer, gl ) {
 			session.addEventListener( 'end', onSessionEnd );
 
 			var attributes = gl.getContextAttributes();
+			if ( ! attributes.xrCompatible ) {
```
Comment (Mugen87) [inline src/renderers/webxr/WebXRManager.js:184]: Minor style issues...

**Comment by toji** [comment]
I'm obviously not the deciding vote here, but this change feels pretty good to me now.

**Comment by mrdoob** [comment]
Sorry, I do not agree with this. I don't understand why users have to know about this.

**Comment by mrdoob** [comment]
Before changing the current API (as this PR does) I want to give my proposal a try: https://github.com/mrdoob/three.js/commit/97144ea3e7ef0697f6243a639cb62158386e0849

**Comment by mrdoob** [comment]
Closing this for now. If my solution creates issues we can revisit this.

**Pull request closed without merging**